### PR TITLE
[draft] CamVid smp segmentation

### PR DESCRIPTION
**Pull request by lzrwch**

### CODE REVIEW

File: training_scripts/segmentation/train_smp.py [hunk 1]
```diff
@@ -0,0 +1,255 @@
+import os
+import numpy as np
+import cv2
+import matplotlib.pyplot as plt
+import segmentation_models_pytorch as smp
+from segmentation_models_pytorch import utils
```
Comment (ahmed-deeplite) [inline training_scripts/segmentation/train_smp.py:6]: where  does`segmentation_models_pytorch` come from?

File: training_scripts/segmentation/train_smp.py [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+DATA_DIR = './data/CamVid/'
+
+# load repo with data if it is not exists
+if not os.path.exists(DATA_DIR):
+    print('Loading data...')
+    os.system('git clone https://github.com/alexgkendall/SegNet-Tutorial .data/')
+    print('Done!')
+
+
+x_train_dir = os.path.join(DATA_DIR, 'train')
```
Comment (ahmed-deeplite) [inline training_scripts/segmentation/train_smp.py:21]: Please follow our conventions for creating dataloaders. Use `get_data_splits_by_name` https://github.com/Deeplite/deeplite-torch-zoo/blob/7c4d9d6dfb16e7e4d6ff43773b2c13de4874af18/deeplite_torch_zoo/wrappers/wrapper.py#L28

File: training_scripts/segmentation/train_smp.py [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+
+x_valid_dir = os.path.join(DATA_DIR, 'val')
+y_valid_dir = os.path.join(DATA_DIR, 'valannot')
+
+x_test_dir = os.path.join(DATA_DIR, 'test')
+y_test_dir = os.path.join(DATA_DIR, 'testannot')
+
+
+
+class Dataset(BaseDataset):
```
Comment (ahmed-deeplite) [inline training_scripts/segmentation/train_smp.py:32]: Please, separate the Dataset files from the training loop

File: training_scripts/segmentation/train_smp.py [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+
+
+ENCODER = 'tu-seresnext50_32x4d'
+ENCODER_WEIGHTS = 'imagenet'
+CLASSES = ['car']
+ACTIVATION = 'sigmoid' # could be None for logits or 'softmax2d' for multiclass segmentation
+DEVICE = 'cuda'
+
+# create segmentation model with pretrained encoder
+model = smp.FPN(
```
Comment (ahmed-deeplite) [inline training_scripts/segmentation/train_smp.py:175]: Please create a separate file for the model

**Review by ahmed-deeplite — CHANGES_REQUESTED** [review]

File: training_scripts/segmentation/train_smp.py (hunk 1 above)
Comment (lzrwch) [reply training_scripts/segmentation/train_smp.py:6]: @honnesh-deeplite add it to setup.py

File: training_scripts/segmentation/train_smp.py [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+            image, mask = sample['image'], sample['mask']
+            
+        return image, mask
+        
+    def __len__(self):
+        return len(self.ids)
+
+# Augmentations using albumentations
+
+def get_training_augmentation():
```
Comment (lzrwch) [inline training_scripts/segmentation/train_smp.py:96]: also place augmentations to a separate file under src/

**Comment by lzrwch** [comment]
@honnesh-deeplite please structure the PR in accordance to zoo structure:

in `deeplite_torch_zoo/src/semantic_segmentation`:
    - in `datasets` folder, place the CamVid and Cityscapes dataset objects in separate files

in `deeplite_torch_zoo/wrappers`:
    - in `datasets/semantic_segmentation` a file with a wrapper fn that creates dataloaders
    - in `eval/semantic_segmentation` a file with a wrapper fn for metrics taken from smp
    - in `eval/models` a file with a wrapper method to create Unet models with custom timm backbones

**Pull request closed without merging**